src/net.cpp

```diff
@@ -574,6 +577,9 @@ bool CNode::Unban(const CSubNet &subNet) {
     LOCK(cs_setBanned);
     if (setBanned.erase(subNet))
     {
+        DumpBanlist(); //store banlist to disk
+        uiInterface.BannedListChanged();
+
         setBannedIsDirty = true;
```

Comment:
Doesn't DumpBanList clear seBannedIsDirty?  I think this flag needs to be set prior to calling DumpBandList, not after.